ntbs-service-unit-tests/DataMigration/NotificationMapperTest.cs

```diff
@@ -277,6 +286,69 @@ public async Task correctlyCreates_PostMortemNotification()
                 te => Assert.Equal(TreatmentOutcomeType.Died, te.TreatmentOutcome.TreatmentOutcomeType));
         }
 
+        // This is based on NTBS-2417
+        [Fact]
+        public async Task correctlyCreates_DeathEventWithoutPostmortem()
+        {
+            // Arrange
+            const int runId = 12345;
+            var legacyIds = new List<string> { "132468" };
+            SetupNotificationsInGroups(("132468", "6"));
+
+            const string colchesterGeneralCode = "TBS0049";
+            _hospitalToTbServiceCodeDict = new Dictionary<Guid, TBService>
+            {
+                {new Guid("0EEE2EC2-1F3E-4175-BE90-85AA33F0686C"), new TBService {Code = colchesterGeneralCode}}
+            };
+
+            // Act
+            var notification = (await _notificationMapper.GetNotificationsGroupedByPatient(null,
+                    runId,
+                    legacyIds))
+                .SelectMany(group => group)
+                .Single();
+
+            // Assert
+            Assert.False(notification.ClinicalDetails.IsPostMortem);
+            Assert.Collection(notification.TreatmentEvents,
+                te => Assert.Equal(TreatmentEventType.DiagnosisMade, te.TreatmentEventType),
+                te => Assert.Equal(16, te.TreatmentOutcomeId),
+                te => Assert.Equal(TreatmentOutcomeType.Died, te.TreatmentOutcome.TreatmentOutcomeType));
+        }
+
+        // This is based on NTBS-2417
+        [Fact]
+        public async Task doesNotAddDeathEvent_WhenOneAlreadyExists()
+        {
+            // Arrange
+            const int runId = 12345;
+            var legacyIds = new List<string> { "132469" };
+            SetupNotificationsInGroups(("132469", "6"));
+
+            const string colchesterGeneralCode = "TBS0049";
+            _hospitalToTbServiceCodeDict = new Dictionary<Guid, TBService>
+            {
+                {new Guid("0EEE2EC2-1F3E-4175-BE90-85AA33F0686C"), new TBService {Code = colchesterGeneralCode}}
+            };
+
+            // Act
+            var notification = (await _notificationMapper.GetNotificationsGroupedByPatient(null,
+                    runId,
+                    legacyIds))
+                .SelectMany(group => group)
+                .Single();
+
+            // Assert
+            Assert.False(notification.ClinicalDetails.IsPostMortem);
+
+            var treatmentEvents = notification.TreatmentEvents.ToList();
+            Assert.Collection(treatmentEvents,
+                te => Assert.Equal(TreatmentEventType.DiagnosisMade, te.TreatmentEventType),
+                te => Assert.Equal(16, te.TreatmentOutcomeId),
+                te => Assert.Equal(7, te.TreatmentOutcomeId));
```

Comment:
`16` and `7` are magic numbers here - they make it quite difficult to know what you're checking for here

Reply:
I've had a go at making them a bit less magical, but the solution is less than ideal, I'm borrowing from one of the seeding functions (the only place I could find `TreatmentOutcomeId` without using the databse)

Reply:
Hmm, I see what you mean. The other option is just to name them just in this class e.g. have a constant called `tbCausedDeathOutcomeId`. It's less rigorous, but it shows the intent of the code clearly.

The other thing you could do (although this is probably overkill) is introduce an enum for the IDs which could be referenced by the seed data.

I'm happy with any of these options, or even leaving it as it currently is.

Reply:
Yep I've gone with the first option (couple of constants naming these magic numbers), the other one seemed pretty heavy-weight given the problem!